### ISSUE TITLE
Unknown lints trigger when running cargo clippy

### ISSUE THREAD

**Issue by antoyo** [rust-lang/rust-clippy#2043]
Hi.
I know this is supposed to be already fixed, but I still get this error:
```rust
   Compiling mg-settings v0.2.2 (file://.../mg-settings)
    Finished dev [unoptimized + debuginfo] target(s) in 1.4 secs
   Compiling mg-settings v0.2.2 (file://.../mg-settings)
warning: unknown lint: `new_without_default_derive`
   --> src/lib.rs:221:13
    |
221 |     #[allow(new_without_default_derive)]
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: #[warn(unknown_lints)] on by default

    Finished dev [unoptimized + debuginfo] target(s) in 3.35 secs
```
when running `cargo clippy`.
Thanks to fix this issue.

**Comment by oli-obk** [comment]
Huh.... is that with the latest clippy and nightly?

**Comment by antoyo** [comment]
```
rustc 1.22.0-nightly (ca94c75c5 2017-09-10)
clippy = "0.0.158"
```
So yes.

**Comment by oli-obk** [comment]
I just tried to build `mg-setting` with `cargo clippy`, and I'm only getting `useless_let_if_seq` and no other lints and no `unknown lint` warnings

**Comment by Dushistov** [comment]
May be related to #2501, and you need `tests` in your crate to trigger warnings?

**Comment by phansch** [comment]
Going to close this as a duplicate of #2501
